### ISSUE TITLE
Release multiple binaries

### ISSUE THREAD

**Issue by caarlos0** [goreleaser/goreleaser#259]
Add a `binaries` or something like that to the config file.

The idea is in [this comment](https://github.com/goreleaser/goreleaser/issues/49#issuecomment-300814646) by @hellais:

> This is currently the limiting fact for me using goreleaser for my releases. I have a single package where I would like to ship multiple binaries that each are generated from a separate main.go. I would like all the binaries to end up inside of the final .tar.gz so having separate goreleaser.yml files is not an option.
> 
> Could this issue be re-opened or should I file another one?

**Comment by caarlos0** [comment]
This would actually break lots of things. My idea so far is:

```yml
name: MyProject
build:
  binaries:
     - main: ./foo/main.go
        name: foo
     - main: ./bar/main.go
        name: bar
```

So, `build` will lost the options `binary` and `main`. Name templates related to archiving and etc would look into the root `name` property. We would have to change the brew templates to consider multiple binaries inside a single archive.

What do you guys think?

**Comment by jorinvo** [comment]
I think adding `binaries` is a good idea.
But I think we should consider keeping `binary` and `main`, too. 
Since the majority of projects have only one binary, they can keep using the simple version. Also, their builds won't suddenly break.
We can add `binaries` additionally and assert that only one of `binary` or `binaries` can be set at the same time.

**Comment by caarlos0** [comment]
Another option, which would also address #228, is to actually have multiple build sections...

**Comment by jorinvo** [comment]
Yes, that would allow for more flexibility. 
I would still argue, that we should make this additional to the current configuration.
Maybe we can add a `builds` section, which would be a list of the same options as in `build`.

**Comment by caarlos0** [comment]
yes, would actually be easier:

- if `build` is set and `builds` is empty, set the `build` item as an item into `builds`
- iterate `builds` everywhere `build` is used
- would still need to add the `project` property

**Comment by hellais** [comment]
I second the option of having `builds` as it would allow for `binary` specific customisation. I also think it would not be nice to break backward compatibility for the existing users, so I think looking for both `builds` and `build` is wise.

> would still need to add the project property

I am not sure what you mean by this

**Comment by caarlos0** [comment]
@hellais can you take a look at https://github.com/goreleaser/goreleaser/pull/267 ? thanks!

**Comment by caarlos0** [comment]
released on v0.22.0
